sparse/_compressed/compressed.py

```diff
@@ -141,6 +142,24 @@ def __init__(
         if prune:
             self._prune()
 
+    def copy(self, deep=True):
+        """Return a copy of the array.
+
+        Parameters
+        ----------
+        deep : boolean, optional
+            If True (default), the internal coords and data arrays are also
+            copied. Set to ``False`` to only make a shallow copy.
+        """
+        return _copy.deepcopy(self) if deep else _copy.copy(self)
+
+    def _make_shallow_copy_of(self, other):
+        self.data = other.data
+        self.indices = other.indices
+        self.indptr = other.indptr
+        self.compressed_axes = other.compressed_axes
+        super().__init__(other.shape, fill_value=other.fill_value)
```

Comment:
```suggestion
        return self.copy(deep=False)
```

Reply:
I don't quite see how this makes a copy of `other`. I tried this and it makes most of the tests fail.

Reply:
This was a misunderstanding on my part.

Reply:
Gotcha, no worries.

Reply:
Could you try `self.__dict__ = other.__dict__.copy()`?

Reply:
Yeah, definitely. Should I change it for COO as well?